modules/engine/src/index.ts

```diff
@@ -1538,6 +1454,18 @@ export class VectorEngine implements IVectorEngine {
     }
   }
 
+  private async runProtocolMethodWithRetries<T = any>(fn: () => Promise<Result<T, ProtocolError>>, retryCount = 5) {
+    let result: Result<T> | undefined;
+    for (let i = 0; i < retryCount; i++) {
+      result = await fn();
+      if (!result.isError) {
```

Comment:
It's a good idea to differentiate between intermittent errors and deterministic errors. That's probably beyond the scope of this PR but worth considering for the future.

Reply:
Yeah, I agree. We are planning on doing a broader error refactor soon(TM) but for now I think we can retry them all